tests/unit/Core/Content/Sitemap/Provider/CategoryUrlProviderTest.php

```diff
@@ -89,8 +121,141 @@ public function testGetCategoryUrls(): void
         static::assertSame('category/2/detail', $url->getLoc());
     }
 
-    private function initServices(): void
+    public function testEmptyUrlResult(): void
+    {
+        $categoryRowNames = array_keys($this->createCategoryResult());
+        $queryResult = new Result(
+            new ArrayResult($categoryRowNames, []),
+            $this->connection
+        );
+        $this->initServices($queryResult);
+        static::assertNotNull($this->queryBuilder);
+        $context = Generator::generateSalesChannelContext();
+
+        $provider = $this->getCategoryUrlProvider();
+        $urlResult = $provider->getUrls($context, 100, 50);
+
+        $urls = $urlResult->getUrls();
+        static::assertCount(0, $urls);
+    }
+
+    public function testEmptyExcludedUrls(): void
```

Comment:
IMO the test functions are too hard to read. They are too dynamic. 

A good unit test should feel like a real test where you provide data and expect data back which you assert afterwards. This test is designed just to run into each single line of code to increase the coverage.